common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -465,7 +465,10 @@ class @Problem
       saveMessage = response.msg
       if response.success
         @el.trigger('contentChanged', [@id, response.html])
-        @render(response.html, @focus_on_save_notification)
+        @el.find('.notification-save .notification-message').html(saveMessage)
+        @clear_all_notifications()
+        @saveNotification.show()
+        @focus_on_save_notification()
```

Comment:
How would you feel about moving the "show" into "focus_on_notification"? While it is not necessary in the "submit" case, it doesn't seem that it would harm anything, and it simplifies the code somewhat.